core/experiments/manager.php

```diff
@@ -76,6 +80,17 @@ public function add_feature( array $options ) {
 
 		$this->features[ $options['name'] ] = $experimental_data;
 
+		if ( is_admin() ) {
+			$feature_option_key = $this->get_feature_option_key( $options['name'] );
+
+			$on_state_change_callback = function( $old_state, $new_state ) use ( $experimental_data ) {
```

Comment:
It look like the `feature` should be a class that handles his state change and other aspects.